MergerLogic/Batching/Tile.cs

```diff
@@ -83,5 +83,10 @@ public void SetCoords(int z, int x, int y)
             this.Y = y;
             this.Z = z;
         }
+
+        public override string ToString()
+        {
+            return $"z: {this.Z}, x: {this.X}, y: {this.Y}, data size: {this._data.Length}";
```

Comment:
protect against null reference for _data

Reply:
it should never be null as it there is no tile the entire tile will be null

Reply:
ok, it's not a good practice but i'll allow it